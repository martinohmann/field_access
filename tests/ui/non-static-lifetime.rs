use field_access::FieldAccess;

#[derive(FieldAccess)]
pub struct Foo<'a> {
    field: &'a [u8],
}

fn main() {}
