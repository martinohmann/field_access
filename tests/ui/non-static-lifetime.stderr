error[E0478]: lifetime bound not satisfied
 --> tests/ui/non-static-lifetime.rs:3:10
  |
3 | #[derive(FieldAccess)]
  |          ^^^^^^^^^^^
  |
note: lifetime parameter instantiated with the lifetime `'a` as defined here
 --> tests/ui/non-static-lifetime.rs:4:16
  |
4 | pub struct Foo<'a> {
  |                ^^
  = note: but lifetime parameter must outlive the static lifetime
  = note: this error originates in the derive macro `FieldAccess` (in Nightly builds, run with -Z macro-backtrace for more info)

error: lifetime may not live long enough
 --> tests/ui/non-static-lifetime.rs:3:10
  |
3 | #[derive(FieldAccess)]
  |          ^^^^^^^^^^^ cast requires that `'a` must outlive `'static`
4 | pub struct Foo<'a> {
  |                -- lifetime `'a` defined here
  |
  = note: this error originates in the derive macro `FieldAccess` (in Nightly builds, run with -Z macro-backtrace for more info)
